src/components/WalletModal/index.tsx

```diff
@@ -174,6 +177,12 @@ export default function WalletModal({
     }
   }, [walletModalOpen, setWalletView, account])
 
+  useEffect(() => {
+    if (account && account !== previousAccount && walletModalOpen) {
+      toggleWalletModal()
+    }
```

Comment:
would it make sense to also set the previous account here?  I guess I'm asking if there is an edge case where ```acount !=== previousAccount``` and  the previous account is never updated, so the wallet toggles back and forth indefinitely

Reply:
well I guess not indefinitely, but everytime you open it

Reply:
but that would reset previous account on the next render, so it won't lead to an infinite loop. since there will be a new render prior to the next user action

Reply:
ok, makes sense.